### PR TITLE
Handle unexpected default in `MessageMismatch` when changing a value

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
Until this PR we handled only the case when a command wants to clear a field. This PR adds handling the case when a command expects some value, and wants to change it, but discovers a default value.

### CODE REVIEW

File: client/src/main/java/org/spine3/change/MessageMismatch.java [hunk 1]
```diff
@@ -70,6 +70,22 @@ public static ValueMismatch expectedNotDefault(Message expected, int version) {
...
+     * Creates a {@code ValueMismatch} for a command that wanted to <em>change</em> a field value,
+     * but discovered that the field has the default value.
+     *
+     * @param expected the value expected by the command
+     * @param newValue the value the command wanted to set
+     * @param version  the version of the entity in which the mismatch is discovered
+     * @return new {@code ValueMismatch} instance
+     */
+    public static ValueMismatch expectedNotDefault(Message expected, Message newValue, int version) {
+        checkNotNull(expected);
```
Comment (armiol) [inline client/src/main/java/org/spine3/change/MessageMismatch.java]: I thought we treat the method parameters as non-nullable due to the annotation in `package-info.java`. If that's still true, I'd think that `checkNotNull` statements should be removed.

If someone passes `null` values to this method, he will get an NPE (or maybe something even worse) for violating an API contract.

File: server/src/main/java/org/spine3/server/event/enrich/EventEnricher.java [hunk 2]
```diff
@@ -55,16 +55,18 @@
 import static org.spine3.protobuf.Messages.toMessageClass;
 
 /**
- * {@code Enricher} extends information of an event basing on its type and content.
+ * {@code EventEnricher} extends information of an event basing on its type and content.
```
Comment (armiol) [inline server/src/main/java/org/spine3/server/event/enrich/EventEnricher.java]: Have you included the `EventEnricher`-related changes to this PR by a mistake? They just don't seem related to the `Mismatch`es at all.

File: client/src/main/java/org/spine3/change/MessageMismatch.java (hunk 1 above)
Comment (alexander-yevsyukov) [inline client/src/main/java/org/spine3/change/MessageMismatch.java]: We annotate the package to say that the API expects non-nulls. But in the `public` and `protected` API we check for `null`s to prevent accidental errors and their propagation into our code deeper.

File: server/src/main/java/org/spine3/server/event/enrich/EventEnricher.java (hunk 2 above)
Comment (alexander-yevsyukov) [inline server/src/main/java/org/spine3/server/event/enrich/EventEnricher.java]: I wanted to have latest `EventEnricher` in  the code from Projects. That's why this PR is based on earlier works.

File: client/src/main/java/org/spine3/change/MessageMismatch.java (hunk 1 above)
Comment (alexander-yevsyukov) [inline client/src/main/java/org/spine3/change/MessageMismatch.java]: As discussed vocally, let's introduce a separate `NullToleranceTest` class that would check `null` parameters for the utility classes.
